### PR TITLE
PM-4977 Migrate Preferences component

### PR DESCRIPTION
**Pull request by KiruthigaManivannan**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Migrate Preferences component to use the component library.

## Code changes

- **preferences.component.html:** Converted the select , radio button group , checkbox form fields to use its equivalent components in library.
- **preferences.component.ts:** Converted Submit to async arrow function.

## Screenshots

https://github.com/bitwarden/clients/assets/162679756/a452d2d1-2e5f-4546-bc31-dbd1037ab231

### CODE REVIEW

File: apps/web/src/app/settings/preferences.component.html [hunk 1]
```diff
@@ -28,102 +28,87 @@
...
-            formControlName="vaultTimeoutAction"
-          />
-          <label class="form-check-label" for="vaultTimeoutActionLogOut">
-            {{ "logOut" | i18n }}
-            <small>{{ "vaultTimeoutActionLogOutDesc" | i18n }}</small>
-          </label>
-        </div>
-      </div>
+          <bit-label>{{ "logOut" | i18n }}</bit-label>
+          <bit-hint class="tw-text-sm">{{ "vaultTimeoutActionLogOutDesc" | i18n }}</bit-hint>
```
Comment (Hinton) [inline apps/web/src/app/settings/preferences.component.html:48]: **issue:** The `bit-hint` component should provide the desired styling, and we should be able to skip `tw-text-small`. While the font-size differs from the non migrated page, that's by design.

**Review by Hinton — COMMENTED** [review]
@KiruthigaManivannan I did an initial review since I was tagged. A few areas where we can clean up the markup and remove some style overrides.

Per the jira ticket i believe Auth is the designated team, so I've requested a review from the team. Assuming Auth is happy with the PR please do not block this PR waiting for a re-review from me.

File: apps/web/src/app/settings/preferences.component.html (hunk 1 above)
Comment (Hinton) [reply apps/web/src/app/settings/preferences.component.html:48]: Yes, please don't override the font size on hints.

File: apps/web/src/app/settings/preferences.component.html [hunk 2]
```diff
@@ -1,129 +1,108 @@
...
-        <app-vault-timeout-input
-          [vaultTimeoutOptions]="vaultTimeoutOptions"
-          [formControl]="form.controls.vaultTimeout"
-          ngDefaultControl
-        >
-        </app-vault-timeout-input>
-      </div>
+  <p bitTypography="body1">{{ "preferencesDesc" | i18n }}</p>
+  <form [formGroup]="form" [bitSubmit]="submit">
+    <div class="tw-w-1/2">
```
Comment (Hinton) [inline apps/web/src/app/settings/preferences.component.html:6]: **suggestion**: We can lift up the class to the form element and get rid of all the divs.

File: apps/web/src/app/settings/preferences.component.html (hunk 1 above)
Comment (KiruthigaManivannan) [reply apps/web/src/app/settings/preferences.component.html]: @Hinton Removed the class applied on bit-hint

image

is this look fine

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**